File: app/src/main/java/aipova/githubsearch/dao/RepoDao.java
package aipova.githubsearch.dao;

import java.io.IOException;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import aipova.githubsearch.App;
import retrofit2.Response;

/**
 * Created by aipova on 07.02.17.
 */

public class RepoDao {
    private static final List<RepoModel> SAMPLE_REPO_LIST = Arrays.asList(
            new RepoModel("Interesting Repo", "Very interesting project"),
            new RepoModel("Not interesting Repo", "Not so interesting project"));

    public List<RepoModel> getSampleRepoModels() {
        return SAMPLE_REPO_LIST;
    }

    public List<RepoModel> getSampleRepoModelsFromApi() throws IOException {
        Response<RepoSearchResult> result = App.getGitHubApi()
                .getRepositories("todolist", "stars", "desc", "1", "10").execute();
        return result.body() != null && result.body().getItems() != null ? result.body().getItems() : Collections.<RepoModel>emptyList();
    }

    public List<RepoModel> getRepoModelsFromApi(String searchQuery, int page) throws IOException {
        Response<RepoSearchResult> result = App.getGitHubApi()
                .getRepositories(searchQuery, "stars", "desc", String.valueOf(page), "10").execute();
        return result.body() != null && result.body().getItems() != null ? result.body().getItems() : Collections.<RepoModel>emptyList();
    }
}

File: app/src/main/java/aipova/githubsearch/dao/RepoDao.kt
package aipova.githubsearch.dao

import aipova.githubsearch.Injection
import java.io.IOException

/**
 * Created by aipova on 07.02.17.
 */

class RepoDao {

    @Throws(IOException::class)
    fun getRepoModelsFromApi(searchQuery: String, page: Int): List<RepoModel>? {
        val result = Injection.gitHubApi
            .getRepositories(searchQuery, "stars", "desc", page.toString(), "10").execute()
        return if (result.body() != null && result.body()!!.items != null) result.body()!!.items else emptyList()
    }
}
